test/client/js/event-target.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2015-2016, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* global describe it before afterEach */
+'use strict';
+import EventTarget from '../../../public/js/event-target';
+const assert = require('assert');
+const simpleMock = require('simple-mock');
+
+describe('js.event-target', () => {
+  let eventTarget;
+  const callbackA = simpleMock.spy(() => {});
+
+  before(() => {
+    eventTarget = new EventTarget();
+  });
+
+  afterEach(() => {
+    simpleMock.restore();
+    callbackA.reset();
+  });
+
+  it('Fires the correct callback', done => {
+    eventTarget.addEventListener('event-a', callbackA);
+    eventTarget.dispatchEvent({type: 'event-a'});
+    setTimeout(() => {
```

Comment:
Looking at https://mochajs.org/#asynchronous-code and also https://lostechies.com/derickbailey/2012/08/17/asynchronous-unit-tests-with-mocha-promises-and-winjs/ I think the right way to do this is to pass `done` to the callback, and arrange for it to be called.

The problem with removing `setTimeout()` completely is that `dispatchEvent()` might (for some reason) only process events after a timeout, in which case the `done()` has already happened. Mocha apparently imposes its own 2 sec timeout for `done()` to happen, so I think this test should look like:

```js
it("Fires the correct callback 2", done => {
  const expectedEvent = { type: "event-a" };
  eventTarget.addEventListener("event-a", actualEvent => {
    assert.deepEqual(actualEvent, expectedEvent);
    done();
  });
  eventTarget.dispatchEvent(expectedEvent);
});
```

(`before()` should probably also be changed to `beforeEach()` to get a fresh `eventTarget` each time, as I discovered when adding a "Fires the correct callback 2" test...)